src/Foundation/NSObject2.cs

```diff
@@ -186,6 +187,11 @@ struct objc_super {
 			}
 		}
 
+		internal bool RequiresUIThread {
```

Comment:
Good news is that we now have GC metrics we're monitoring in CI, so once we insert it, we can validate how much this improves things.